pallets/inactivity-tracking/src/lib.rs

```diff
@@ -0,0 +1,370 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+//! # Inactivity Tracking Pallet
+//!
+//! This pallet tracks and stores the activity of container chain and orchestrator chain collators
+//! for configurable number of sessions. It is used to determine if a collator is inactive
+//! for that period of time.
+//!
+//! The tracking functionality can be enabled or disabled with root privileges.
+//! By default, the tracking is enabled.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use {
+    frame_support::{dispatch::DispatchResult, pallet_prelude::Weight},
+    parity_scale_codec::{Decode, Encode},
+    scale_info::TypeInfo,
+    serde::{Deserialize, Serialize},
+    sp_core::{MaxEncodedLen, RuntimeDebug},
+    sp_runtime::{traits::Get, BoundedBTreeSet},
+    sp_staking::SessionIndex,
+    tp_traits::{
+        AuthorNotingHook, AuthorNotingInfo, GetContainerChainsWithCollators, GetSessionIndex,
+        MaybeSelfChainBlockAuthor, NodeActivityTrackingHelper,
+    },
+};
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+pub mod weights;
+pub use weights::WeightInfo;
+
+#[cfg(feature = "runtime-benchmarks")]
+use tp_traits::{BlockNumber, ParaId};
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use tp_traits::ForSession;
+    use {
+        super::*,
+        crate::weights::WeightInfo,
+        core::marker::PhantomData,
+        frame_support::{pallet_prelude::*, storage::types::StorageMap},
+        frame_system::pallet_prelude::*,
+        sp_std::collections::btree_set::BTreeSet,
+    };
+
+    /// The status of the activity tracking
+    #[derive(
+        Clone,
+        PartialEq,
+        Eq,
+        Encode,
+        Decode,
+        TypeInfo,
+        Serialize,
+        Deserialize,
+        RuntimeDebug,
+        MaxEncodedLen,
+    )]
+    pub enum ActivityTrackingStatus {
+        Enabled {
+            start: SessionIndex,
+            end: SessionIndex,
+        },
+        Disabled {
+            end: SessionIndex,
```

Comment:
I don't understand what are start and end, can you add doc comments?

Reply:
Added doc comments explaining each  status parameter value means in https://github.com/moondance-labs/tanssi/pull/922/commits/19d28a1b6304312438cbbe172f00917c85f8e01e